tests/e2e/prevent_duplicate_cluster_ids.go

```diff
@@ -0,0 +1,474 @@
+		defer cancel()
+
+		framework.Logf("CNS_TEST: Running for vanilla k8s setup")
+
+		ginkgo.By("Creating csi config secret with no cluster id field set")
+		recreateVsphereConfigSecret(client, ctx, vCenterUIUser, vCenterUIPassword, csiNamespace, vCenterIP,
+			"", vCenterPort, dataCenter, csiReplicas)
+
+		ginkgo.By("Verify cluster id configmap is auto generated by csi driver")
+		verifyClusterIdConfigMapGeneration(client, ctx, csiNamespace, true)
+		clusterID := fetchClusterIdFromConfigmap(client, ctx, csiNamespace)
+		framework.Logf("clusterID: %v", clusterID)
+
+		ginkgo.By("Creating Storage Class")
+		if rwxAccessMode {
+			scParameters[scParamFsType] = nfs4FSType
+		}
+		sc, err := createStorageClass(client, scParameters, nil, "", "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			ginkgo.By("Delete Storage Class")
+			err = client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating service")
+		service := CreateService(namespace, client)
+		defer func() {
+			deleteService(namespace, client, service)
+		}()
+		// Check if it is file volumes setups
+		if rwxAccessMode {
+			accessMode = v1.ReadWriteMany
+		}
+		ginkgo.By("Creating statefulset with replica 3 and a deployment")
+		statefulset, deployment, _ := createStsDeployment(ctx, client, namespace, sc, true,
+			false, 0, "", accessMode)
+		replicas := *(statefulset.Spec.Replicas)
+
+		defer func() {
+			scaleDownNDeleteStsDeploymentsInNamespace(ctx, client, namespace)
+			pvcs, err := client.CoreV1().PersistentVolumeClaims(namespace).List(ctx, metav1.ListOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			for _, claim := range pvcs.Items {
+				pv := getPvFromClaim(client, namespace, claim.Name)
+				err := fpv.DeletePersistentVolumeClaim(client, claim.Name, namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				ginkgo.By("Verify it's PV and corresponding volumes are deleted from CNS")
+				err = fpv.WaitForPersistentVolumeDeleted(client, pv.Name, poll,
+					pollTimeout)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				volumeHandle := pv.Spec.CSI.VolumeHandle
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(volumeHandle)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+					fmt.Sprintf("Volume: %s should not be present in the CNS after it is deleted from "+
+						"kubernetes", volumeHandle))
+			}
+		}()
+
+		// Scale up replicas of statefulset and verify CNS entries for volumes
+		scaleUpStsAndVerifyPodMetadata(ctx, client, namespace, statefulset,
+			replicas+2, true, true)
+		verifyVolumeMetadataOnDeployments(ctx, client, deployment, namespace, nil, nil,
+			nil, "")
+		err = checkClusterIdValueOnWorkloads(&e2eVSphere, client, ctx, namespace, clusterID)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		scaleDownNDeleteStsDeploymentsInNamespace(ctx, client, namespace)
```

Comment:
Does this util method delete PVC and PV after sts Pods deletion or scale down ?

If No, Can we do a cleanup of PVC,Pod also hereafter sts scale down?